aten/src/ATen/native/cuda/ConvolutionMM2d.cu

```diff
@@ -117,17 +117,13 @@ void slow_conv2d_forward(
            const Tensor &weight_,
            const Tensor &bias,
            const Tensor &columns,
-           const Tensor &ones_,
            int64_t kH, int64_t kW,
            int64_t dH, int64_t dW,
            int64_t padH, int64_t padW) {
   auto weight = new_view_weight_MM2d(weight_);
   slow_conv2d_shape_check(
       input, {}, weight, bias, kH, kW, dH, dW, padH, padW, /*weight_nullable*/false);
 
-  TORCH_CHECK(!bias.defined() || bias.is_contiguous(),
-              "bias tensor has to be contiguous");
-
```

Comment:
Just to verify my understading: `bias` no longer needs to be contiguous because `grad_bias` is now being computed using `at::sum_out` instead of with an ad-hoc kernel that required contiguous input?

Reply:
That's the general idea. This is replacing cuBLAS library calls which don't support completely arbitrary strides, with `TensorIterator` kernels that do.